src/NetAnalyzers/CSharp/Microsoft.NetCore.Analyzers/Performance/CSharpDoNotGuardDictionaryRemoveByContainsKeyFixer.cs

```diff
@@ -16,11 +17,11 @@ public sealed class CSharpDoNotGuardDictionaryRemoveByContainsKeyFixer : DoNotGu
     {
         protected override bool SyntaxSupportedByFixer(SyntaxNode conditionalSyntax)
         {
-            if (conditionalSyntax is ConditionalExpressionSyntax conditionalExpressionSyntax)
-                return conditionalExpressionSyntax.WhenTrue.ChildNodes().Count() == 1;
-
-            if (conditionalSyntax is IfStatementSyntax)
-                return true;
+            if (conditionalSyntax is IfStatementSyntax ifStatementSyntax)
+            {
+                return ifStatementSyntax.Condition.RawKind != (int)SyntaxKind.LogicalNotExpression &&
```

Comment:
Consider adding a comment why we are skipping the case where the condition is a logical not expression.

Reply:
Also added one to the VB fixer.